### ISSUE TITLE
[Design Doc] Reward Function Refinement: From Linear to EIG-based

### ISSUE THREAD

**Issue by h315uk3** [h315uk3/symbiosis#44]
## Purpose
Document the design refinement of the question reward function from a linear additive model to an information-theoretically grounded EIG (Expected Information Gain) based approach.

### Implementation
```python
reward = (0.40 * info_gain + 
          0.20 * clarity +
          0.20 * specificity +
          0.10 * relevance +
          0.10 * completeness)
```

### Problems

**1. Independence Assumption (Critical)**
- Formula assumes all components are independent
- In reality: clarity ⊥̸ specificity (correlated)
  - Clearer questions tend to be more specific
  - No justification for linear superposition

**2. No Theoretical Grounding**
- Weights (0.40, 0.20, ...) are arbitrary
- No principled method to adjust weights
- 'info_gain' is not formally defined (word-count proxy)

**3. Inflexible to New Dimensions**
- Adding new component requires weight recalibration
- Entire formula breaks if one component changes

**4. Double Counting**
- clarity, specificity, relevance all capture similar concepts
- Redundant information penalizes multi-faceted questions

---

### Approach 1: Bayesian Hierarchical Model
**Concept:** Model reward as f(EIG, clarity) with learned dependency structure

**Pros:**
- Captures component correlations formally
- Principled uncertainty quantification

**Cons:**
- Requires statistical training data
- Stdlib-only constraint → infeasible (no scipy.stats)

---

### Approach 2: Purely EIG-based (Information-Theoretic)
**Concept:** Reward = EIG(Q) solely, no secondary components

**Pros:**
- Maximally principled (grounded in information theory)
- No arbitrary weights

**Cons:**
- Ignores other legitimate quality dimensions (clarity, specificity)
- May reward technically-high-value but poorly-written questions

---

### Approach 3: Hybrid EIG + Adjustments ⭐ SELECTED

**Concept:**
```
r(Q) = EIG(Q) + 0.1×clarity(Q) + 0.05×importance(Q)
```

**Rationale for Each Component:**

1. **Primary: EIG(Q)** (Information Gain in bits)
   - Directly measures reduction in uncertainty
   - Theoretically grounded
   - Primary decision criterion

2. **Secondary: 0.1 × clarity(Q)** (Quality adjustment)
   - Questions should be understandable
   - Prevents value being assigned to unclear questions
   - Weight 0.1: secondary to EIG, but significant

3. **Tertiary: 0.05 × importance(Q)** (Strategic weighting)
   - Some dimensions more important than others
   - Can be customized per deployment
   - Weight 0.05: minor strategic adjustment

---

## Why Hybrid (案3) Over Pure EIG (案2)

**Scenario: Principle vs. Practice**

Consider two questions Q1, Q2 with same EIG=1.5 bits:

**Q1 (Well-written):**
- EIG = 1.5 bits
- clarity = 0.8 (good)
- importance = 1.0 (important dimension)
- r = 1.5 + 0.1×0.8 + 0.05×1.0 = **1.63**

**Q2 (Unclear):**
- EIG = 1.5 bits
- clarity = 0.2 (confusing)
- importance = 0.5 (less important)
- r = 1.5 + 0.1×0.2 + 0.05×0.5 = **1.55**

**Outcome:** Q1 ranked higher (reasonable) while preserving EIG primacy

**Pure EIG Would Rank Equally:** Both would be 1.5, losing quality signal

---

### Primary Weight: EIG has coefficient 1.0
- Carries full information-theoretic signal
- All uncertainty reduction is valued equally

### Secondary Weight: clarity has coefficient 0.1
- 10% of potential EIG value
- Max adjustment: 0.1 (if clarity=1.0)
- Prevents low-quality questions dominating despite high EIG

### Tertiary Weight: importance has coefficient 0.05
- 5% of potential EIG value
- Max adjustment: 0.05 (if importance=1.0)
- Strategic tuning without dominating EIG signal

### Why These Specific Values?

1. **Additive, not multiplicative:**
   - Multiplicative would amplify low-clarity questions to near-zero
   - Additive preserves EIG as primary signal

2. **Decreasing weights:**
   - EIG >> clarity >> importance (diminishing returns)
   - Reflects information-theoretic primacy

3. **Small secondary weights:**
   - 0.1 and 0.05 are intentionally small
   - Prevent non-information-theoretic factors dominating
   - Allow fine-tuning without major reward shifts

---

### Phase 1: EIG Core
- Implement dimension_belief.py with HypothesisSet
- Calculate EIG per dimension
- Baseline reward: r = EIG

### Phase 2: Quality Adjustments
- Add clarity estimation from question text
- Add importance weighting by dimension
- Adjust r = EIG + 0.1×clarity + 0.05×importance

### Phase 3 (Future): Learning
- If data accumulates, learn optimal weights via regression
- Current: manual specification (sound rationale)
- Future: empirical optimization if needed

---

## Doctest Migration

**Old Tests:**
```python
>>> evaluate_question('What is the purpose?')
{'reward': 0.65, 'info_gain': 0.2, 'clarity': 0.25, ...}
```

**New Tests:**
```python
>>> evaluate_question('What is the purpose?')
{'reward': 1.15, 'eig': 1.1, 'clarity': 0.5, 'importance': 1.0, ...}
```

**Strategy:**
- Update all 47 doctests to reflect new reward structure
- Link to design rationale in docstrings
- Add comments explaining weight choices

---

## Trade-offs Summary

| Aspect | Trade-off | Decision | Rationale |
|--------|-----------|----------|-----------|
| **Theoretical Purity** | EIG-only vs. Hybrid | Hybrid | Practicality wins small battles |
| **Weight Learning** | Manual vs. Learned | Manual (Phase 1) | Rationale-driven; learn later if needed |
| **Component Count** | Minimal vs. Comprehensive | 3 components | Sweet spot between simplicity & practicality |
| **Scaling** | Additive vs. Multiplicative | Additive | Preserves EIG primacy |

---

## Related Issues
- #37: Phase 1-A (dimension_belief.py)
- #38: Phase 1-B (EIG reward function implementation)
- #43: Theoretical foundation (Bayesian approach)
